### PR TITLE
tool/ceph-dedup-tool: improve the progress report

### PR DESCRIPTION
**Pull request by myoungwon**
This PR improves the progress reports during chunk scrub and sample dedup.

Signed-off-by: Myoungwon Oh <myoungwon.oh@samsung.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by myoungwon** [comment]
@athanatos Can you take a look? The goal of this PR is to improve progress report.

File: src/tools/ceph_dedup_tool.cc
Comment (athanatos) [inline src/tools/ceph_dedup_tool.cc:575]: maybe_print_status()

File: src/tools/ceph_dedup_tool.cc
Comment (athanatos) [inline src/tools/ceph_dedup_tool.cc:618]: const utime_t

File: src/tools/ceph_dedup_tool.cc
Comment (athanatos) [inline src/tools/ceph_dedup_tool.cc:620]: const uint32_t report_period -- can you express the default where the option is defined?

**Review by athanatos — CHANGES_REQUESTED** [review]

File: src/tools/ceph_dedup_tool.cc
Comment (athanatos) [inline src/tools/ceph_dedup_tool.cc:463]: const

**Review by athanatos — CHANGES_REQUESTED** [review]

File: src/tools/ceph_dedup_tool.cc
Comment (myoungwon) [reply src/tools/ceph_dedup_tool.cc]: Done.

File: src/tools/ceph_dedup_tool.cc
Comment (myoungwon) [reply src/tools/ceph_dedup_tool.cc]: Done.

File: src/tools/ceph_dedup_tool.cc
Comment (myoungwon) [reply src/tools/ceph_dedup_tool.cc]: Moved to the options_description. Please see the last commit.

File: src/tools/ceph_dedup_tool.cc
Comment (myoungwon) [reply src/tools/ceph_dedup_tool.cc]: Done.

**Review by athanatos — APPROVED** [review]

**Comment by myoungwon** [comment]
@athanatos Can this be merged? or need QA. I think this just related to prints.

**Comment by athanatos** [comment]
It needs qa.  Apologies, I should have tagged it when I approved it.  In general, all PRs must go through their respective test suites.  Isn't there a dedup test merged in the rados suite now?

**Comment by myoungwon** [comment]
Correct. Tests related to ceph_dedup_tool are in` qa/workunits/rados/test_dedup_tool.sh`.
Tests for rados operations (e.g., set_chunk, flush, snapshot and so on) are in `qa/suites/rados/thrash/workloads/`.
Specifically, 
`qa/suites/rados/thrash/workloads/dedup-io-mixed.yaml`
`qa/suites/rados/thrash/workloads/dedup-io-snaps.yaml`
`qa/suites/rados/thrash/workloads/set-chunks-read.yaml`
`qa/suites/rados/thrash/workloads/redirect*.yaml`

**Comment by athanatos** [comment]
Great.  In general, the requirements to merge a PR related to dedup are an approval and a successful rados suite run.

**Comment by athanatos** [comment]
@myoungwon Are you able to schedule a run?

**Comment by athanatos** [comment]
Should go through qa with https://github.com/ceph/ceph/pull/52261

**Comment by myoungwon** [comment]
I'm waiting for receiving an re-approval of credentials to access sepia, because my laptop is new (https://tracker.ceph.com/issues/63033#change-247466). After getting the approval, I can schedule the run.

**Comment by myoungwon** [comment]
@athanatos I cherry-pick the commits in this PR and schedule the run with https://github.com/ceph/ceph/pull/52261.
The tests are complete. There are no concerns I think.
https://pulpito.ceph.com/myoungwon-2023-10-15_06:36:14-rados-wip-ceph-dedup-tool-snap-add-distro-default-smithi/
https://pulpito.ceph.com/myoungwon-2023-10-23_00:47:41-rados-wip-ceph-dedup-tool-snap-add-distro-default-smithi/

**Comment by myoungwon** [comment]
@athanatos ping.

**Pull request merged**
